src/System.Linq.Expressions/tests/Unary/UnaryQuoteTests.cs

```diff
@@ -47,5 +48,461 @@ public void QuotePreservesTypingOfBlock(bool useInterpreter)
 
         public static Type Quote1(Expression<Action> e) => e.Body.Type;
         public static Type Quote2(Expression<Func<object>> e) => e.Body.Type;
+
+#if FEATURE_COMPILE
+        [Fact]
+        public static void Quote_Lambda_Action()
```

Comment:
The current limitation is documented in https://github.com/dotnet/corefx/pull/11314, causing the interpreter and compiler to emit different expressions for runtime quotes. I'll have another look at that now.